src/observer.ts

```diff
@@ -8,31 +8,67 @@ export function useStaticRendering(enable: boolean) {
     isUsingStaticRendering = enable
 }
 
-export function observer<P>(baseComponent: FunctionComponent<P>): FunctionComponent<P> {
+export interface IObserverOptions {
+    readonly forwardRef?: boolean
+}
+
+export function observer<P extends object, TRef = {}>(
+    baseComponent: React.RefForwardingComponent<TRef, P>,
+    options: IObserverOptions & { forwardRef: true }
+): React.MemoExoticComponent<
+    React.ForwardRefExoticComponent<React.PropsWithoutRef<P> & React.RefAttributes<TRef>>
+>
+export function observer<P extends object>(
+    baseComponent: React.FunctionComponent<P>,
+    options?: IObserverOptions
+): React.NamedExoticComponent<P>
+
+// n.b. base case is not used for actual typings or exported in the typing files
+export function observer<P extends object, TRef = {}>(
+    baseComponent: React.RefForwardingComponent<TRef, P>,
+    options?: IObserverOptions
+) {
+    const realOptions = {
```

Comment:
Move this under the check for static rendering please.

Reply:
done